docs/training/apis-howto/api-core-mnist-ug.rst

```diff
@@ -0,0 +1,573 @@
+evaluation loop. In addition, pass the newly created ``core_context`` into both ``train()`` and
+``test()``.
+
+.. code::
+
+   # NEW: Pass core_context into train() and test().
+   train(args, model, device, train_loader, optimizer, epoch, core_context)
+
+   # NEW: Pass args, test_loader, epoch, and steps_completed into test().
+   test(args, model, device, test_loader, epoch, core_context, steps_completed=steps_completed)
+
+Create a ``steps_completed`` variable to plot metrics on a graph in the WebUI:
+
+.. note::
+
+   Refer to line 198 in ``model_def_metrics.py``
+
+.. code::
+
+   # NEW: Calculate steps_completed for plotting test metrics.
+   steps_completed = epoch * len(train_loader)
+
+Step 2.4: Run the Experiment
+============================
+
+Run the following command to run the experiment:
+
+Finally, run the following command to run the experiment:
+
+.. code::
+
+   det e create metrics.yaml .
+
+Open the Determined WebUI again and navigate to the **Overview** tab.
+
+The WebUI now displays metrics. In this step, you learned how to add a few new lines of code in
+order to report training and validation metrics to the Determined master. Next, we’ll modify our
+script to report checkpoints.
+
+***********************
+ Step 3: Checkpointing
+***********************
+
+Checkpointing periodically during training and reporting the checkpoints to the master gives us the
+ability to stop and restart training. In this section, we’ll modify our script for the purpose of
+checkpointing.
+
+..
+   note::Two Types of Continuations (Stopping and Restarting)
+
+   There are two types of continuations (methods for stopping and restarting
+   training): pausing and reactivating training using the WebUI or clicking
+   **Continue Trial** after the experiment completes.
+
+   Each type of continuation has its own behavior. While you always want to preserve
+   the value you are incrementing (the “model weight”), you do not always
+   want to preserve the batch index. When you pause and reactivate you want
+   training to continue from the same batch index. However, when starting a
+   fresh experiment, you want training to start with a fresh batch index.
+
+   To distinguish between the two types of continuations, you can save the
+   trial ID in the checkpoint.
+
+In this step, we’ll run our experiment using the ``model_def_checkpoints.py`` script and its
+accompanying ``checkpoints.yaml`` experiment configuration file.
+
+Step 3.1: Save Checkpoints
+==========================
+
+To save checkpoints, add the ``store_path`` function to your script:
+
+.. note::
+
+   Refer to line 208 in ``model_def_checkpoints.py``
+
+.. code::
+
+   # NEW: Save checkpoint.
+     checkpoint_metadata_dict = {"steps_completed": steps_completed}
+
+     with core_context.checkpoint.store_path(checkpoint_metadata_dict) as (path, storage_id):
+       torch.save(model.state_dict(), path / "checkpoint.pt")
+
+.. image:: /assets/images/model-def-checkpoints-script.png
```

Comment:
why do we have both snippets and screenshots of code?

Reply:
removed screenshots